homeassistant/components/sensor/yr.py

```diff
@@ -0,0 +1,251 @@
+                         hass.config.time_zone, 0))
+
+    google = GoogleGeocoder()
+    try:
+        google._get_elevation(location)  # pylint: disable=protected-access
+        _LOGGER.info(
+            'Retrieved elevation from Google: %s', location.elevation)
+        elevation = location.elevation
+    except urllib.error.URLError:
+        # If no internet connection available etc.
+        elevation = 0
+
+    coordinates = dict(lat=hass.config.latitude,
+                       lon=hass.config.longitude, msl=elevation)
+
+    weather = YrData(coordinates)
+
+    dev = []
+    if 'monitored_conditions' in config:
+        for variable in config['monitored_conditions']:
+            if variable not in SENSOR_TYPES:
+                _LOGGER.error('Sensor type: "%s" does not exist', variable)
+            else:
+                dev.append(YrSensor(variable, weather))
+
+    # add symbol as default sensor
+    if len(dev) == 0:
+        dev.append(YrSensor("symbol", weather))
+    add_devices(dev)
+
+
+# pylint: disable=too-many-instance-attributes
+class YrSensor(Entity):
+    """ Implements an Yr.no sensor. """
+
+    def __init__(self, sensor_type, weather):
+        self.client_name = 'yr'
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self.type = sensor_type
+        self._state = None
+        self._weather = weather
+        self._info = ''
+        self._unit_of_measurement = SENSOR_TYPES[self.type][1]
+        self._update = datetime.datetime.fromtimestamp(0)
+
+        self.update()
+
+    @property
+    def name(self):
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """ Returns state attributes. """
+        data = {}
+        data[''] = self._info + "Weather forecast from yr.no, delivered by the"\
+            " Norwegian Meteorological Institute and the NRK"
+        if self.type == 'symbol':
+            symbol_nr = self._state
+            data[ATTR_ENTITY_PICTURE] = "http://api.met.no/weatherapi/weathericon/1.1/" \
+                                        "?symbol=" + str(symbol_nr) + \
+                                        ";content_type=image/png"
+
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement of this entity, if any. """
+        return self._unit_of_measurement
+
+    @property
+    def should_poll(self):
+        """ Return True if entity has to be polled for state. """
+        return True
+
+    # pylint: disable=too-many-branches, too-many-return-statements
+    def update(self):
+        """ Gets the latest data from yr.no and updates the states. """
+
+        self._weather.update()
+        now = datetime.datetime.now()
+        # check if data should be updated
+        if now > self._update:
```

Comment:
Not a comment to fix but merely a tip for the future: consider using guard clauses to make the code more readable (less-indentations).

Instead of this:
```python
if now > self._update:
    […]
```

Write this:
```python
if now <= self._update:
    return

[…]
```

In this file it could be applied to line 164, 179 (using `continue`), 236.